### PR TITLE
Simplify the example crates

### PR DESCRIPTION
**Pull request by sunfishcode**
- Rename test-crates to example-crates
 - Change from a .cargo/config.toml to a build.rs, as that doesn't require --target.
 - Mention that origin-start is how origin-studio works.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,20 +43,18 @@ Origin can also be used on its own, in several different configurations:
...
-   start the program using its own program entrypoint. This version must be
-   compiled with an explicit `--target=` flag, even if it's the same as the
-   host, because it uses special RUSTFLAGS, and an explicit `--target` flag
-   prevents those flags from being passed to build scripts. origin handles
-   program and thread startup and shutdown and no part of libc is used.
+   start the program using its own program entrypoint. origin handles program
+   and thread startup and shutdown and no part of libc is used. This is the
+   approach that [origin-studio] uses.
 
  - The [origin-start-no-alloc example] is like origin-start, but disables the
```
Comment (morr0ne) [inline README.md:50]: `origin-start-no-alloc example` doesn't actually link to anything, I'm assuming it was just a little oversight and not intentional?

**Comment by morr0ne** [comment]
Guess this is too late since it got merged while I was writing this comment. But regardless, while trying to resolve #23 I figured out that `nostartfiles` is not compatible with lld, using the build.rs method would make the examples incompatible with the changes I proposed there

**Comment by sunfishcode** [comment]
I'm open to adding an example that defines its own target too, but I don't expect everyone who uses origin to use it that way, so my thought is to have both examples side by side.

**Comment by morr0ne** [comment]
> I'm open to adding an example that defines its own target too, but I don't expect everyone who uses origin to use it that way, so my thought is to have both examples side by side.

Having both examples seem like the best solution, I am a bit worried about the whole static linking problem however. One of my uses cases for origin involves running in environments with nothing more than the kernel itself, that is pretty much impossible unless a custom target is used so I feel like documenting that executable compiled with origin are not actually static is appropriate. On a first glance they look static but its a rather nasty surprise when you deploy to a machine that requires static executable and the program doesn't run.

**Pull request merged**